docs/lockservice.md

```diff
@@ -0,0 +1,61 @@
+# Lock Service
+
+This document describes the design and functionality of the lock service.
+
+## High Level Functionality
+The lock service implements a synchronization mechanism that can be used by other applications. It implements functions for acquiring and releasing locks to objects present in the application. The lock to an object is leased for a certain duration to the caller. This is done to prevent starvation of other requests to the lock. When the entity that holds the lock is done using the corresponding object, it can release the lock. If the lock's lease expires then it will be released and assigned to the next waiting entity that wishes to acquire it.
+
+Clients communicate with the lock server via HTTP. `node` contains code for a HTTP server. When this is started, incoming requests are routed to their corresponding functions in the lock service using route handlers defined in `route`.
+
+## Low-level implementation
+The lockservice uses the following data structure to keep track of lock acquisitions: 
+```go
+type SafeLockMap struct {
+	LockMap map[string]string
+	Mutex   sync.Mutex
+}
+```
+On an `Acquire()` or `Release()`, the mutex of `SafeLockMap` is locked. The LockMap stores a mapping of the object that is locked to the sessionID that currently owns that object.
```

Comment:
Mention something like - "The mutex is locked so that there are no concurrent accesses"; this isn't clear.

Reply:
done!